src/object.c

```diff
@@ -239,7 +239,9 @@ void pthreads_current_thread(zval *return_value) {
 
 /* {{{ */
 int pthreads_connect(pthreads_object_t* source, pthreads_object_t* destination) {
-	if (source && destination) {
+	pthreads_ident_t destCreator = destination->creator;
```

Comment:
Can you place this declaration inside of the `if` statement below, please. The below condition is checking to make sure `destination` is not `NULL`, so there must be some case when it can be `NULL` (in which case, the current code will segfault by trying to deref a `NULL` pointer (`destination`)).